.circleci/config.yml

```diff
@@ -39,17 +90,25 @@ jobs:
              make defconfig > /dev/null
 
       - run:
-          name: Download & check & compile
+          name: Download source/check/compile
           working_directory: ~/build_dir
           command: |
-             PKGS=$(cd ~/openwrt_packages; git diff --diff-filter=d --name-only "origin/$BRANCH..." | grep 'Makefile$' | grep -v '/files/' | awk -F/ '{ print $(NF-1) }')
-             echo "Packages: $PKGS"
+             PKGS=$(cd ~/openwrt_packages; git diff --diff-filter=d --name-only "$COMMIT_RANGE" | grep 'Makefile$' | grep -v '/files/' | awk -F/ '{ print $(NF-1) }')
+             if [ -z "$PKGS" ] ; then
+                 echo_blue "WARNING: No new or modified packages found!"
+                 exit 0
+             fi
+
+             echo_blue "=== Found new/modified packages: $PKGS"
              for PKG in $PKGS ; do
+                 echo_blue "===+ Download/check: $PKG"
                  make "package/$PKG/download" V=s
                  make "package/$PKG/check" V=s
              done
+
              for PKG in $PKGS ; do
-                 make "package/$PKG/compile" -j3 V=s
+                 echo_blue "===+ Building: $PKG"
+                 make "package/$PKG/compile" -j$(getconf _NPROCESSORS_ONLN) V=s
```

Comment:
Maybe we should print `getconf _NPROCESSORS_ONLN` because CPU detection in container is not always perfect

Reply:
Yes, you found the one thing I didn't verify explicitly - will do

Reply:
root@fdbfea96c4de:~# getconf _NPROCESSORS_ONLN
36

Reply:
Maybe you can burst a bit but don't think they let you use 36 vCPU when they advertise 2